dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/annotations/MatcherCreateStart.java

```diff
@@ -0,0 +1,29 @@
+package com.dat3m.dartagnan.program.event.core.annotations;
+
+public class MatcherCreateStart extends CodeAnnotation {
```

Comment:
We could fully remove it and use simply a skip. That might be a better option than making this more general when we don't have another use case.

Reply:
I would want to have something descriptive. Maybe just a very generic `StringAnnotation` event that is essentially a `skip` but with associated text.
Then you would use `EventFactory.newStringAnnotation("Spawning thread " + tId/name)` which would be printed like this:
```
===== Spawning thread X =====
Fence;
store(...)
Fence;

```

Reply:
Ok. But instead of creating yet another class which will simply extend `Skip`, with not add a label argument to `Skip`, return the label when calling `toString()` and in the factory allow `newSkip(label)` and `newSkip()` which will simply return `new Skip("skip")`

Reply:
Well, the event would still extend `CodeAnnotation` and hence inherit the `Annotation` tag.
I think it makes sense to distinguish between plain skip events and code annotations.

Reply:
Ok, but does it make sense to keep code annotations abstract and keep adding events that extend it? So far only FunCall and FunRet extend it and both have a string, so we can simply make the string part of the CodeAnnotation and use this event for the matcher (and simplify the other two classes). Then CodeAnnotation would have the tag and the string annotation we want to use in the code, and FunRet and FunCall just a minimal logic about the function name which is used in the simplification pass

Reply:
I would keep `CodeAnnotation` abstract because not every annotation needs a string (we also have a set of loop annotations none of which take strings as input).
But it might make sense to create a single subclass that does nothing but hold a string (`StringAnnotation` or `Comment`?). We might then use this class whenever we just want to place a generic marker or inherit from it for specialized string-based markers (like `FunCall` and `FunRet`).